File: backend/Invoices.Application/ReadModels/Product/ProductPagedList.cs
using System;
using System.Linq;
using System.Linq.Expressions;
using Domain.Pagination;

namespace Invoices.Application.ReadModels.Product
{
    public class ProductPagedList : PagedList<ProductView>
    {
        public Guid Id { get; set; }
        public decimal NetToPay { get; set; }
        public decimal GrossToPay { get; set; }
        public ProductPagedList(Guid id, decimal netToPay, decimal grossToPay,
            IQueryable<ProductView> items, int pageSize, int currentPage, Expression<Func<ProductView, string>> orderByExpression) : base(items, pageSize, currentPage, orderByExpression)
        {
            Id = id;
            NetToPay = netToPay;
            GrossToPay = grossToPay;
        }
    }
}

File: frontend/invoices-frontend/src/app/invoices/models/product/productpagedlist.ts
import { ProductView } from './productview';

export class ProductPagedList{
id: string;
  netToPay: number;
  grossToPay: number;
  totalItems: number;
  currentPage: number;
  pageSize: number;
  totalPages: number;
  startIndex: number;
  endIndex: number;
  items: ProductView[];
}